lambda_alerts/main.tf

```
@@ -13,28 +15,53 @@ variable "alarm_actions" {
 }
 
 variable "error_rate_threshold" {
+  type        = number
   description = "The threshold error rate (as a percentage) for triggering an alert"
   default     = 1
 }
 
+variable "memory_usage_threshold" {
+  type        = number
+  description = "The threshold memory utilization (as a percentage) for triggering an alert"
+  default     = 90
+}
+
 variable "datapoints_to_alarm" {
+  type        = number
   description = "The number of datapoints that must be breaching to trigger the alarm."
   default     = 1
 }
 
 variable "evaluation_periods" {
+  type    = number
   default = 1
 }
 
 variable "period" {
+  type        = number
   description = "The period in seconds over which the specified statistic is applied."
   default     = 60
 }
 
 variable "treat_missing_data" {
+  type    = string
   default = "missing"
 }
 
+locals {
+
+  insights_layer_name = "LambdaInsightsExtension"
+
+  layers = data.aws_lambda_function.target.layers
+
+  insights_enabled = anytrue([for layer in local.layers : strcontains(layer, local.insights_layer_name)])
+
+}
```

Just to tighten this up format-wise (and to please `terraform fmt`), I'd recommend:
```suggestion
locals {
  insights_layer_name = "LambdaInsightsExtension"
  insights_enabled = anytrue([
    for layer in local.layers : strcontains(layer, local.insights_layer_name)
  ])
  layers = data.aws_lambda_function.target.layers
}

---

Format changes have been incorporated!